src/core/reaction_ensemble.cpp

```diff
@@ -831,6 +831,12 @@ bool ReactionAlgorithm::do_global_mc_move_for_particles_of_type(
     p_id = p_id_s_changed_particles[i];
     // change particle position
     new_pos = get_random_position_in_box();
+    double vel[3];
```

Comment:
Can you please add:
auto const &p = get_particle_data(p_id);
vel[0] = std::sqrt(temperature/p.p.mass) * gaussian_random();
vel[1] = std::sqrt(temperature/p.p.mass) * gaussian_random();
vel[2] = std::sqrt(temperature/p.p.mass) * gaussian_random();
etc in this place? This way the particle moving MC moves give correctly thermalized particle velocities.

Reply:
Can we use the same to get the charges?

Reply:
Here we can ask for the particle mass, since it was already created by the user beforehand (with the correct mass). In the other location we cannot ask for the particle mass or charge, since the particle is about to be created with a certain user provided mass or charge. Espresso cannot know what charge or mass should be assigned if a new particle is to be created (since this is physical input).

Reply:
So I incorporated the changes. Is there more to do, or is this pull request ready to be merged?